Tfm update for new build system

ioannisg opened the pull request:
Required fixes in Zephyr in the wake of upgrading TF-M to the new build system (+ getting support for nRF devices).

The upmerge is present in: https://github.com/zephyrproject-rtos/trusted-firmware-m/pull/15

The following platforms can currently build and run the TFM Integration samples successfully:
- mps2_an521_nonsecure
- nrf5340pdk_nrf5340_cpuappns
- nrf9160dk_nrf9160ns
- nucleo_l552ze_q_ns

The following platforms now also build successfully (but I have no HW to test)
- lpcxpresso55s69
- musca_b1

The PR also contains the required doc changes for samples and boards.

ioannisg committed:
boards: arm: update board definitions due to new TF-M build system

TF-M related information in the relevant board definitions
needs to be updated in the wake of the new TF-M Build
system.

Signed-off-by: Ioannis Glaropoulos <Ioannis.Glaropoulos@nordicsemi.no>

ioannisg committed:
modules: trusted-firmware-m: update default RSA keys

Update the default paths to private keys used
for signing the Secure and the Non-Secure firmware,
when building Zephyr together with TF-M. The update
is done to match the default configuration in
the upstream TF-M project.

Signed-off-by: Ioannis Glaropoulos <Ioannis.Glaropoulos@nordicsemi.no>

ioannisg committed:
boards: arm: update tfm post-build scripts after upmerge

Update the TF-M post-build scripts in the relevant
ARM platforms in the wake of updating TF-M to the
latest upstream.

Signed-off-by: Ioannis Glaropoulos <Ioannis.Glaropoulos@nordicsemi.no>

ioannisg committed:
boards: arm: Add support for TF-M in nRF5340 PDK

Adding support for TF-M in the Nordic nRF5340
Application MCU.

Signed-off-by: Ioannis Glaropoulos <Ioannis.Glaropoulos@nordicsemi.no>

ioannisg committed:
manifest: update trusted-firmware-m to latest revision

Update manifest file for tf-m to get the latest
revision in Zephyr.

Signed-off-by: Ioannis Glaropoulos <Ioannis.Glaropoulos@nordicsemi.no>

erwango commented on boards/arm/nucleo_l552ze_q/CMakeLists.txt:
```diff
@@ -27,36 +28,36 @@ if (CONFIG_BUILD_WITH_TFM)
+			 -s auto
 			 -H 0x400
-			 ${CMAKE_BINARY_DIR}/tfm/install/outputs/STM_NUCLEO_L552ZE_Q/tfm_s.bin
+			 ${CMAKE_BINARY_DIR}/tfm/install/outputs/MPS2/AN521/tfm_s.bin
```
I guess this change is not correct, but I'm not sure if it should be `stm/nucleo_l552ze_q` of  the existing line was fine

agansari replied:
Looks like we don't actually need custom vendor/board paths, binaries are copied to ./build/tfm/bin/
This could be used further to unify the merge script.

ioannisg replied:
That's probably true @agansari, we can revisit this as part of the cleanup (something along what @anangl had done in https://github.com/zephyrproject-rtos/zephyr/pull/28806)

I'd just fix this for now, since I want to separate the upmerge from the cleanup 👍

ioannisg replied:
This is fixed, now, @erwango

ioannisg committed:
boards: arm: Add support for TF-M in nRF9160 DK

Adding support for TF-M in the Nordic nRF9160 DK.
Allow the TF-M integration samples to be built and
executed for nRF9160 DK.

Signed-off-by: Ioannis Glaropoulos <Ioannis.Glaropoulos@nordicsemi.no>

tejlmand commented on boards/arm/mps2_an521/CMakeLists.txt:
```diff
@@ -18,7 +18,8 @@ if (CONFIG_BUILD_WITH_TFM)
 	endif()
 
-	set(PREPROCESSED_FILE "${CMAKE_BINARY_DIR}/tfm/image_macros_preprocessed")
+	set(PREPROCESSED_FILE_S "${CMAKE_BINARY_DIR}/tfm/bl2/ext/mcuboot/CMakeFiles/signing_layout_s.dir/signing_layout_s.o")
+	set(PREPROCESSED_FILE_NS "${CMAKE_BINARY_DIR}/tfm/bl2/ext/mcuboot/CMakeFiles/signing_layout_ns.dir/signing_layout_ns.o")
```
I believe those two files should be be returned by the TF-M build, or exists as TF-M build properties that can be fetched when needed.

This is a bit to deep knowledge of the TF-M build structure.

tejlmand replied:
can of course be improved later, just wanted to mention it.

tejlmand commented on boards/arm/nrf5340dk_nrf5340/CMakeLists.txt:
```diff
@@ -7,3 +7,73 @@ if (((CONFIG_BOARD_NRF5340PDK_NRF5340_CPUAPP OR CONFIG_BOARD_NRF5340PDK_NRF5340_
+		set(TFM_PUBLIC_KEY_FORMAT "full")
+	endif()
+
+	# Set srec_cat binary name
+	find_program(SREC_CAT srec_cat)
+	if(${SREC_CAT} STREQUAL SREC_CAT-NOTFOUND)
+	    message(FATAL_ERROR "'srec_cat' not found. Please install it, or add it to $PATH.")
+	endif()
```
I don't see `srec_cat` being used here ?

tejlmand commented on boards/arm/nrf9160dk_nrf9160/CMakeLists.txt:
```diff
@@ -5,3 +5,73 @@ if(CONFIG_BOARD_NRF52840_GPIO_RESET)
+		set(TFM_PUBLIC_KEY_FORMAT "full")
+	endif()
+
+	# Set srec_cat binary name
+	find_program(SREC_CAT srec_cat)
+	if(${SREC_CAT} STREQUAL SREC_CAT-NOTFOUND)
+	    message(FATAL_ERROR "'srec_cat' not found. Please install it, or add it to $PATH.")
+	endif()
```
I don't see `srec_cat` being used here.

tejlmand commented on samples/tfm_integration/psa_level_1/boards/nrf5340pdk_nrf5340_cpuappns.overlay:
```diff
@@ -0,0 +1,32 @@
+			reg = <0x20040000 DT_SIZE_K(192)>;
+		};
+	};
+};
```
seems to lack a newline.

tejlmand commented on samples/tfm_integration/psa_level_1/boards/nrf9160dk_nrf9160ns.overlay:
```diff
@@ -0,0 +1,32 @@
+			reg = <0x20040000 DT_SIZE_K(192)>;
+		};
+	};
+};
```
seems to lack a newline.

tejlmand reviewed (changes requested):
General impression is ok.
I see some repetition like everyone having to set `TFM_TARGET_PLATFORM` (which are almost aligned with Zephyr board folder name) and `TFM_PUBLIC_KEY_FORMAT.
And there are probably more.
Should we consider (later) to have a kind of Zephyr board -> TFM settings table, instead of each having to write the same CMake code ?

Also I think we need to look into a more proper handling of TF-M output files, but such cleanup can be done after this PR.

ioannisg committed:
boards: arm: add documentation for integrating TF-M on nRF platforms

Add the required documentation extensions for the nRF5340
and the nRF9160 DKs that explains how to build and run
Zephyr with TF-M as the Secure firmware image counterpart.

Signed-off-by: Ioannis Glaropoulos <Ioannis.Glaropoulos@nordicsemi.no>

ioannisg committed:
CODEOWNERS: add code owners for the TF-M integration samples

Adding code owners for the TF-M integration samples.

Signed-off-by: Ioannis Glaropoulos <Ioannis.Glaropoulos@nordicsemi.no>

ioannisg commented:
@carlescufi @anangl @oyvindronningstad could you take a look at [this ](https://github.com/zephyrproject-rtos/zephyr/pull/29027/commits/7f80039772ea260a6503d7d33b88c99b8557568b) commit (docs)

ioannisg commented on samples/tfm_integration/tfm_ipc/boards/nrf9160dk_nrf9160ns.overlay:
```diff
@@ -0,0 +1,32 @@
+ * statically in the TF-M project.
+ */
+&sram0_s {
+	reg = <0x20000000 DT_SIZE_K(192)>;
```
Agreed, although this does not really play any role, at the moment, right? Since tFM has it's own build process that does not take DTS into account....

ioannisg replied:
done

anangl commented on boards/arm/nrf5340dk_nrf5340/doc/index.rst:
```diff
@@ -200,6 +209,33 @@ The process requires the following steps:
+      * Build the Non-Secure firmware image as a regular Zephyr application
+      * Build a TF-M (secure) firmware image
+      * Optionally build a bootloader image (MCUboot)
+      * Merge the output binaries together
```
I'm not sure if it will be clear that only the two firmware images are merged together and the bootloader one is not. Maybe it would be better to change the order of the last two points?

ioannisg replied:
Yes! that's good point 👍

ioannisg replied:
done, and also for nrf9160

ioannisg committed:
boards: nucleo_l552ze_q_ns: pass compiler name to the post-build script

Pass the compiler name and path to the TF-M post-build
script as an argument, so the TF-M build can work with
the Zephyr SDK as well (not only with the GNU ARM embedded
toolchain.)

Signed-off-by: Ioannis Glaropoulos <Ioannis.Glaropoulos@nordicsemi.no>

ioannisg commented on boards/arm/nrf5340dk_nrf5340/CMakeLists.txt:
```diff
@@ -7,3 +7,83 @@ if (((CONFIG_BOARD_NRF5340PDK_NRF5340_CPUAPP OR CONFIG_BOARD_NRF5340PDK_NRF5340_
+
+	set(PREPROCESSED_FILE_S "${CMAKE_BINARY_DIR}/tfm/bl2/ext/mcuboot/CMakeFiles/signing_layout_s.dir/signing_layout_s.o")
+	set(PREPROCESSED_FILE_NS "${CMAKE_BINARY_DIR}/tfm/bl2/ext/mcuboot/CMakeFiles/signing_layout_ns.dir/signing_layout_ns.o")
+	set(TFM_MCUBOOT_DIR "${ZEPHYR_BASE}/../modules/tee/tfm/trusted-firmware-m/bl2/ext/mcuboot")
```
sure, done (for all platforms)

ioannisg commented on samples/tfm_integration/psa_level_1/boards/nrf9160dk_nrf9160ns.overlay:
```diff
@@ -0,0 +1,32 @@
+/ {
+	reserved-memory {
+		sram0_ns: image_ns@20040000 {
+			reg = <0x20040000 DT_SIZE_K(192)>;
```
done

ioannisg commented on samples/tfm_integration/psa_level_1/boards/nrf9160dk_nrf9160ns.overlay:
```diff
@@ -0,0 +1,32 @@
+ * statically in the TF-M project.
+ */
+&sram0_s {
+	reg = <0x20000000 DT_SIZE_K(256)>;
```
done

anangl commented on samples/tfm_integration/psa_level_1/sample.yaml:
```diff
@@ -5,7 +5,8 @@ sample:
     sample.tfm_ipc:
         tags: introduction
-        platform_allow: mps2_an521_nonsecure lpcxpresso55s69_ns
+        platform_allow: mps2_an521_nonsecure lpcxpresso55s69_ns nrf5340pdk_nrf5340cpuappns
```
```suggestion
        platform_allow: mps2_an521_nonsecure lpcxpresso55s69_ns nrf5340pdk_nrf5340_cpuappns
```

ioannisg replied:
good catch, fixed now, thanks.

erwango commented on samples/tfm_integration/tfm_ipc/boards/nucleo_l552ze_q_ns.overlay:
```diff
@@ -28,18 +31,18 @@
 			label = "mcuboot";
-			reg = <0x00000000 0x00011000>;
+			reg = <0x00000000 0x00013000>;
 			read-only;
 		};
 		/* Secure image primary slot */
-		slot0_partition: partition@00011000 {
+		slot0_partition: partition@00013000 {
 			label = "image-0";
-			reg = <0x00011000 0x00040000>;
+			reg = <0x00013000 0x00038000>;
 		};
 		/* Non-secure image primary slot */
-		slot1_partition: partition@00051000 {
+		slot1_partition: partition@0004B000 {
 			label = "image-1";
-			reg = <0x00051000 0x0002E000>;
+			reg = <0x0004B000 0x0002A000>;
```
Is that required ?
If yes, I think I'd be better moved in dedicated commit anyway

ioannisg replied:
There is already a dedicated commit, @erwango concerning this, in the trusted-firmware-m repository.
https://github.com/zephyrproject-rtos/trusted-firmware-m/pull/15/commits/8dca035c06a23907998c2c550f55fdc68136bc86

ioannisg replied:
So this is required, because the TFM layout is changed now (or, strictly speaking , it will be changed if the upstream fixes get merged). I can move this to a separate commit, no problem.

ioannisg replied:
moved, now.

erwango commented on boards/arm/nucleo_l552ze_q/CMakeLists.txt:
```diff
@@ -16,49 +16,50 @@ if (CONFIG_BUILD_WITH_TFM)
+			 -s auto
 			 -H 0x400
-			 ${CMAKE_BINARY_DIR}/tfm/install/outputs/STM_NUCLEO_L552ZE_Q/tfm_s.bin
+			 ${CMAKE_BINARY_DIR}/tfm/install/outputs/STM/NUCLEO_L552ZE_Q/tfm_s.bin
```
Should is be lower case here to match TFM_TARGET_PLATFORM value ?

ioannisg replied:
Nope, this should be compliant with upstream TFM - this is how it looks like in the new build system.

erwango replied:
ok, txs

erwango commented on boards/arm/nucleo_l552ze_q/CMakeLists.txt:
```diff
@@ -60,6 +66,6 @@ if (CONFIG_BUILD_WITH_TFM)
 
 		#Execute post build script postbuild.sh
-		COMMAND ${CMAKE_BINARY_DIR}/tfm/postbuild.sh
+		COMMAND ${CMAKE_BINARY_DIR}/tfm/postbuild.sh ${COMPILER_FULL_PATH}
```
This is a nice improvement.
Please also update the following note: https://github.com/zephyrproject-rtos/zephyr/blob/3ce7afdaaa005dc0f46d721ad77d0c275ce0df9f/samples/tfm_integration/tfm_ipc/README.rst#L132

ioannisg replied:
done :)

erwango reviewed (changes requested):
Few comments

ioannisg commented on boards/arm/mps2_an521/CMakeLists.txt:
```diff
@@ -35,43 +36,43 @@ if (CONFIG_BUILD_WITH_TFM)
 			 --public-key-format ${TFM_PUBLIC_KEY_FORMAT}
 			 --align 1
 			 -v ${TFM_IMAGE_VERSION_NS}
+			 -s auto
 			 ${ADD_S_IMAGE_MIN_VER}
 			 ${ADD_SECURITY_COUNTER_NS}
```
yeap, just that I missed removing it. It does not affect the code, but it is misleading.
Removed now.

Actually, we should be allowing the user to set this e.g. via Kconfig and pass this to both the tf-m build and/or the post-build signing utility....

ioannisg committed:
samples: tfm_integration: adapt CMakeLists.txt to new TF-M build system

Changes required in TF-M integration samples CMakeLists.txt
files due to the new upstream TF-M build system. In brief,
configuration related to PSA API (IPC), Regression, and
Isolation level needs to be passed, now, directly, not
via invoking a given TF-M CMake configuration.

Signed-off-by: Ioannis Glaropoulos <Ioannis.Glaropoulos@nordicsemi.no>

ioannisg committed:
samples: tfm_integration: tfm_ipc: adapt nucleo_l552ze_q_ns overlay

This commit adapts the DTS overlay for nucleo_l552ze_q_ns
board, to comply with the changes applied in the flash
layout of the TF-M build.

Signed-off-by: Ioannis Glaropoulos <Ioannis.Glaropoulos@nordicsemi.no>

ioannisg committed:
samples: tfm_integration: fix build & flash guides for all platforms

Add documentation on how to build and run the TF-M integration
samples on nRF platforms.

The arm-none-eabi-gcc does not need to be on the PATH anymore
for building and running the tfm_ipc sample on nucleo so remove
the note from the sample docs.

Fix some paths for TFM binaries for the LPC board as well.

Signed-off-by: Ioannis Glaropoulos <Ioannis.Glaropoulos@nordicsemi.no>

ioannisg committed:
.buildkite: temporary fix to manually install imgtool

Force CI to manually install imgtool, which is needed
for sanitycheck to build and run the TF-M integration
samples with MCUboot (where MCUboot is built from the
TF-M external project).

Signed-off-by: Ioannis Glaropoulos <Ioannis.Glaropoulos@nordicsemi.no>

mbolivar-nordic commented on .buildkite/run.sh:
```diff
@@ -53,6 +53,10 @@ ccache -s
 # lpcxpresso11u68 boards
 pip3 install lpc_checksum
 
+# Temporary fix: Install imgtool, needed for MCUboot to sign images
+# when builindg the TF-M integration samples
+pip3 install imgtool
```
Can't you put this in one of the pip install lines in .github/foo.yaml instead?

ioannisg replied:
In that case, I suppose it should be in `./github/workflows/sanitycheck_tests.yml`, but I am not sure if this would be better that having this temporary hack in .buildkite (which could be eventually removed, once we have a new image for CI).
Could you elaborate?

@galak any ideas (since you initially suggested this)?

mbolivar-nordic replied:
> Could you elaborate?

We have a bunch of different pip install lines throughout the github workflows which is already confusing for people and causing problems.

For example, I had to send https://github.com/zephyrproject-rtos/zephyr/pull/28922 because https://github.com/zephyrproject-rtos/zephyr/pull/28047 passed CI by editing one pip install line in one workflow used by CI, but the docs daily build used a different environment. I don't like the situation already, it's messy and people don't know where to put pip install lines anymore as seen from this example, and adding yet another place where we have to remember they might be found is not making things better.

ioannisg replied:
FIne, I will try with adding this, then, in ./github/workflows/sanitycheck_tests.yml if this is OK with you.

mbolivar-nordic replied:
> FIne, I will try with adding this, then, in ./github/workflows/sanitycheck_tests.yml if this is OK with you.

Sounds good, at least it doesn't add a new place to find pip install lines. Thanks.

ioannisg replied:
done now, @mbolivar-nordic , hope this does not break CI

ioannisg replied:
Turns out, this won't work, @mbolivar-nordic, as this workflow activates only upon changes directly on sanity check scripts. I don't think it makes sense to add a new workflow here - perhaps better to revert to the initial workaround (patching .buildkite), as there is already a similar workaround there (for `lpc_checksum`).
I can file an enhancement issue to track the removal of this workaround.

mbolivar-nordic replied:
>  I can file an enhancement issue to track the removal of this workaround.

Please do, thanks.

mbolivar-nordic commented on scripts/requirements-build-test.txt:
```diff
@@ -22,3 +22,6 @@ mypy
 
 # used for mocking functions in pytest
 mock>=4.0.1
+
+# used in MCUboot for signing and key management
+imgtool>=1.6.0
```
Not sure if the answer is yes or no, but can you check if cmake/mcuboot.cmake support for finding imgtool via PATH or from mcuboot/scripts should be extracted instead of making this a generic requirement?

ioannisg replied:
The problem is that MCUboot in this case, @mbolivar-nordic , is built via TF-M, which is an external project. It is not build with Zephyr.

ioannisg replied:
...in other words,  cmake/mcuboot.cmake, is not processed at all.

mbolivar-nordic replied:
> ...in other words, cmake/mcuboot.cmake, is not processed at all.

Yes, of course, I'm asking if factoring out that code to find imgtool into shared code would be a better approach. That is why I said "extracted".

mbolivar-nordic replied:
By putting this here it looks like you are basically forcing everybody who uses zephyr+mcuboot+signing via the build system to get imgtool from pip instead of from the mcuboot repository. When I added mcuboot.cmake, it was a deliberate choice to favor imgtool installed via pip over the one found in the mcuboot repository because imgtool was **not** installed by default in the zephyr requirements.txt. This is changing that default.

ioannisg replied:
OK, so to conclude, what would be your suggestion @mbolivar-nordic ?

I cannot, really, hack the external project to look for imgtool from anywhere else than PIP; that will be a nightmare to maintain. I am OK with removing it from requirements.txt, since I already added a doc entry in the TFM samples regarding the python requirements for building TFM binaries (FYI @microbuilder ) so maybe we can afford to skip adding this here...

No strong opinion, but I would also like @microbuilder to comment.

mbolivar-nordic replied:
> OK, so to conclude, what would be your suggestion @mbolivar-nordic ?

My suggestion is to find imgtool in host-tools.cmake by extracting the code in mcuboot.cmake and setting an IMGTOOL CMake variable to something you can run to run imgtool.

My point is that if imgtool is a host tool that is needed in multiple places now (as this PR seems to be making the case), it should probably be treated like any other host tool in the sense of its discovery getting moved to host-tools.cmake. Previously I didn't do this because mcuboot.cmake was a special purpose use case.

cc @tejlmand

ioannisg replied:
@mbolivar-nordic based on @microbuilder 's comment below, we 'll remove this edit in the requirements-build-test.txt.

mbolivar-nordic replied:
> we 'll remove this edit in the requirements-build-test.txt.

Thanks.

ioannisg committed:
samples: tfm_integration: fix requirements for building TFM binaries

In the samples' documentation we update the requirements
for building TF-M binaries according to the latest upstream
TF-M documentation.

Signed-off-by: Ioannis Glaropoulos <Ioannis.Glaropoulos@nordicsemi.no>

ioannisg commented:
DNM until the manifest pointer is updated.

erwango commented on samples/tfm_integration/psa_level_1/sample.yaml:
```diff
@@ -5,7 +5,8 @@ sample:
 tests:
     sample.tfm_ipc:
```
Not related to current change, but might deserve a fix

ioannisg replied:
Fixing that too, thanks.

microbuilder commented:
@ioannisg Concerning the imgtool dependency, I'm OK to only have this requirement documented as mandatory for TF-M, since there are other requirements to meet anyway, rather than forcing the install on everyone for this edge case.

ioannisg commented:
> @ioannisg Concerning the imgtool dependency, I'm OK to only have this requirement documented as mandatory for TF-M, since there are other requirements to meet anyway, rather than forcing the install on everyone for this edge case.

Thanks. Then I'll remove the edit in the requirements_.txt

ioannisg committed:
samples: tfm_integration: psa_level_1: fix sample name in .yaml

Fix the name of the sample in the .yaml file so it matches
the sample directory, and does not collide with tfm_ipc.

Signed-off-by: Ioannis Glaropoulos <Ioannis.Glaropoulos@nordicsemi.no>

ioannisg committed:
samples: tfm_integration: tfm_ipc: add nucleo_l552ze_q_ns to the list

Add nucleo_l552ze_q_ns board to the allowed list of platforms
for the tfm_ipc sample. This allows the sample to be built for
the nucleo_l552ze_q_ns platform by sanity check.

Signed-off-by: Ioannis Glaropoulos <Ioannis.Glaropoulos@nordicsemi.no>

ioannisg commented:
@tejlmand please, take a look so this review could be finalized.

mbolivar-nordic reviewed (approved):
My feedback is addressed.

ioannisg commented:
> My feedback is addressed.

Yeap, thanks.

tejlmand commented on samples/tfm_integration/psa_level_1/README.rst:
```diff
@@ -140,13 +140,46 @@ and non-secure (``zephyr.hex``) images wth a J-Link as follows:
+
+      cd <ZEPHYR_ROOT>/samples/tfm_integration/psa_level_1/
+      rm -rf build
+      mkdir build && cd build
+      cmake -GNinja -DBOARD=nrf9160dk_nrf9160ns ..
```
Just wondering, why not just do:
```suggestion
      cmake -GNinja -DBOARD=nrf9160dk_nrf9160ns -Bbuild .
```
that will create the `build` folder and thus save the `mkdir builld && cd build` step

Note, this is not blocking in this PR, as I know this description has been used elsewhere.

ioannisg replied:
Yeap, we sort of use the old description in most of the cases, so while this is a good suggestion, I'd prefer not to implement it for this particular sample (but we could apply tree-wide later)

tejlmand replied:
yeap. Fully accepted, just wanted to point it out.

tejlmand commented on samples/tfm_integration/tfm_ipc/CMakeLists.txt:
```diff
@@ -14,20 +14,19 @@ elseif (CONFIG_TFM_BL2_FALSE)
 
 # Add "tfm" as an external project via the TF-M module's cmake file
-if(${TFM_TARGET_PLATFORM} STREQUAL "STM_NUCLEO_L552ZE_Q")
+if(${TFM_TARGET_PLATFORM} STREQUAL "stm/nucleo_l552ze_q")
```
Just noticing, using `${TFM_TARGET_PLATFORM}` without qoutes is dangerous in case `TFM_TARGET_PLATFORM` has not been defined.
Will give an error like:
```
CMake Error at samples/tfm_integration/tfm_ipc/CMakeLists.txt:17 (if):
  if given arguments:

    "STREQUAL" "stm/nucleo_l552ze_q"

  Unknown arguments specified
```
whereas qouting would be comparing an empty string, which would be a valid comparison.
```suggestion
if("${TFM_TARGET_PLATFORM}" STREQUAL "stm/nucleo_l552ze_q")
```
and of cause be false, if `TFM_TARGET_PLATFORM` os empty/undefined.

Note, as this is old code (only changing right side string), then it has probably not been a problem up til noew and can thus be cleaned up later.

ioannisg replied:
Thanks @tejlmand , this is good feedback. I can actually fix this in the current PR.

ioannisg replied:
Fixed now.

tejlmand reviewed (approved):
Changed approved.

I still think the TFM integration needs an overhaul / cleanup in general.
It was rushed in for 2.3, and haven't gotten the required cleanup afterwards, but thas is outside the scope of this PR.
